Detectors/Vertexing/src/SVertexer.cxx

```diff
@@ -1090,60 +1096,72 @@ int SVertexer::check3bodyDecays(const V0Index& v0Idx, const V0& v0, float rv0, s
     tr0.getPxPyPzGlo(p0);
     tr1.getPxPyPzGlo(p1);
     tr2.getPxPyPzGlo(p2);
-    std::array<float, 3> p3B = {p0[0] + p1[0] + p2[0], p0[1] + p1[1] + p2[1], p0[2] + p1[2] + p2[2]};
-
-    float pt2candidate = p3B[0] * p3B[0] + p3B[1] * p3B[1], p2candidate = pt2candidate + p3B[2] * p3B[2];
-    if (pt2candidate < mMinPt23Body) { // pt cut
-      continue;
-    }
-    if (p3B[2] * p3B[2] / pt2candidate > mMaxTgl23Body) { // tgLambda cut
-      continue;
+    for (int i = 0; i < 3; i++) {
+      p0[i] /= tr0.getAbsCharge();
+      p1[i] /= tr1.getAbsCharge();
+      p2[i] /= tr2.getAbsCharge();
```

Comment:
Please do this in the loop on the 3 body hypotheses and only for the bachelor tracks